x/amm/keeper/msg_server_create_pool.go

```diff
@@ -19,6 +21,23 @@ func (k msgServer) CreatePool(goCtx context.Context, msg *types.MsgCreatePool) (
 	sender := sdk.MustAccAddressFromBech32(msg.Sender)
 	params := k.GetParams(ctx)
 
+	if params.EnableBaseCurrencyPairedPoolOnly {
+		baseCurrency, found := k.assetProfileKeeper.GetUsdcDenom(ctx)
+		if !found {
+			return nil, errorsmod.Wrapf(assetprofiletypes.ErrAssetProfileNotFound, "asset %s not found", ptypes.BaseCurrency)
+		}
+
+		is_base_curency_paired_pool := false
+		for _, asset := range msg.PoolAssets {
+			if asset.Token.Denom == baseCurrency {
+				is_base_curency_paired_pool = true
+			}
+		}
+		if !is_base_curency_paired_pool {
+			return nil, errorsmod.Wrapf(types.ErrOnlyBaseCurrencyPoolAllowed, "one of the asset must be %s", ptypes.BaseCurrency)
+		}
+	}
+
```

Comment:
@cryptokage1996 can you add a unit test that checks if given the new param enabled and if the user attempts to create a pool without using the base currency that it will result with the expected error. thanks

Reply:
missing tests were added to the PR